### ISSUE TITLE
Feature request: integrating snapraid-btrfs

### ISSUE THREAD

**Issue by goofrider** [OpenMediaVault-Plugin-Developers/openmediavault-snapraid#40]
Would you consider integrating [snapraid-btrfs](https://github.com/automorphism88/snapraid-btrfs)? It uses btrfs snapshots for snapraid sync, which prevents write hole problems.

**Comment by Max-Sum** [comment]
I don't think we could use the script directly but the idea could be implemented.
Now snapshot support is integrated into OMV core. We can now assume that sharefolder = subvolume and there's no nested subvolume in sharefolder. So basically what we need to do is before sync, make a snapshot for each sharefolder on the data disks.

**Comment by goofrider** [comment]
Thank you for your response and glad to hear the idea is being considered. 🙏

**Comment by microraptor** [comment]
There was work done integrating the script here: https://github.com/CCraigen/openmediavault-snapraid-btrfs

However, as you said integrating the functionality more natively into OMV without the script is most likely the best way. This plugin with BTRFS would then be quite unique in solving usability issues other parity have.

**Comment by ryecoaaron** [comment]
>  We can now assume that sharefolder = subvolume and there's no nested subvolume in sharefolder. So basically what we need to do is taking a snapshot for each sharefolder on the data disks before sync.

There are a few problems with this.

- Other filesystems are still supported
- The shared folder on btrfs was created with the new OMV btrfs support.
- You only use btrfs
- You only have data in shared folders since the plugin currently works at the disk level not sharedfolder level

> There was work done integrating the script here:

I looked at that but it is very outdated now.  It basically just creates a snapper config file and swaps the snapraid command out for snapraid-btrfs.  Since snapper would not be used in this proposal, there isn't much work that can be used.

> as you said integrating the functionality more natively into OMV without the script is most likely the best way.

I agree but it would be a huge amount of work especially to keep working with other filesystems and the majority of OMV snapraid uses are not using btrfs.

It probably should be a different plugin that works at the sharedfolder level and only supports btrfs.

**Comment by microraptor** [comment]
> It probably should be a different plugin that works at the sharedfolder level and only supports btrfs.

That does seem to make a lot of sense.

Just for clarification to other readers: A separate BTRFS plugin on shared-folder level (instead of disk level) is necessary, because the new shared folders are automatically subvolumes (which enables the snapshot feature). SnapRAID excludes any nested subvolumes and therefore all shared folders on BTRFS, which were created the new way. See the first posts of this discussion: https://forum.openmediavault.org/index.php?thread/47536-is-snapraid-compatable-with-new-btrfs-snapshots/
